### PR TITLE
Fix PreferredCode function to better fit the necessary specs

### PR DESCRIPTION
**Pull request by ljades**
PreferredCode did not previously account for the ValuesetOid attached to the specific Entry, found in the MapDataCriteria. The new function, and tests surrounding it, does so properly.

In this PR is also a modification to uses of the _result_value partial template, which also requires the MapDataCriteria in its scope.

### CODE REVIEW

File: models/coded.go [hunk 1]
```diff
@@ -86,16 +86,18 @@ func computeIntersection(a []string, b []string) []string {
...
 	if len(c.Codes) == 0 {
 		return Concept{}
 	}
+	mdcValueSet := vsMap[mdcOid]
 	if valueSetPreferred {
 		for codeSystem := range c.Codes {
 			for _, code := range c.Codes[codeSystem] {
 				for _, vsOid := range preferredCodeSets {
-					valueSet := vsMap[vsOid]
-					if len(valueSet) == 0 || codeSetContainsCode(valueSet, CodedConcept{CodeSystem: codeSystem, Code: code}) {
```
Comment (dczulada) [inline models/coded.go:98]: if (len(prefValueSet) == 0 || codeSetContainsCode(prefValueSet, CodedConcept{CodeSystem: codeSystem, Code: code})) && codeSetContainsCode(mdcValueSet, CodedConcept{CodeSystem: codeSystem, Code: code}) {

The Code ALWAYS needs to be in the mdcValueSet.  The original OR placement does not enforce that.

Comment (ljades) [reply models/coded.go:98]: Whoops, that is definitely a typo. Good catch.

**Review by dczulada — APPROVED** [review]
Appears to be working soundly

**Comment by tstrass** [comment]
@sylinmino just needs a rebase

**Pull request merged**